### ISSUE TITLE
Error after python3 upgrade on CEL 1.1.1

### ISSUE THREAD

**Issue by chrisheacock** [OctoPrint/OctoPrint-CuraEngineLegacy#7]
Greetings!

After an upgrade to python 3.7 on OctoPrint, I'm getting this error when slicing. It looks like a lot of python3 work was done recently, but I didn't see anything specific to this, I tried a few .encode()s but couldn't get anything to cooperate.

```
  File "/home/pi/oprint/lib/python3.7/site-packages/octoprint_curalegacy/__init__.py", line 330, in do_slice
    used_extruders=1)
  File "/home/pi/oprint/lib/python3.7/site-packages/octoprint_curalegacy/__init__.py", line 511, in _convert_to_engine
    return profile.convert_to_engine(used_extruders=used_extruders)
  File "/home/pi/oprint/lib/python3.7/site-packages/octoprint_curalegacy/profile.py", line 932, in convert_to_engine
    "endCode": self.get_gcode("end_gcode", extruder_count=actual_extruder_count),
  File "/home/pi/oprint/lib/python3.7/site-packages/octoprint_curalegacy/profile.py", line 767, in get_gcode
    + '\n' + postfix).strip() + '\n'
  File "/home/pi/oprint/lib/python3.7/re.py", line 192, in sub
    return _compile(pattern, flags).sub(repl, string, count)
  File "/home/pi/oprint/lib/python3.7/site-packages/octoprint_curalegacy/profile.py", line 733, in replaceTagMatch
    return pre + 'CURA_OCTO_PROFILE_STRING:%s' % (self.get_profile_string())
  File "/home/pi/oprint/lib/python3.7/site-packages/octoprint_curalegacy/profile.py", line 718, in get_profile_string
    return base64.b64encode(zlib.compress("\b".join(result), 9))
```

I reinstalled the plugin, just to see if anything would change and it behaves the same.

Thanks for any help!

**Comment by markwal** [comment]
@chrisheacock It's particular to your start/end gcode.  Could you upload your profile somewhere and post a link to it here?

**Comment by chrisheacock** [comment]
http://paste.debian.net/1173919/

Thanks!

**Comment by markwal** [comment]
I'm sorry Chris, looks like I didn't get round to it quick enough to save it from expiring.  If you like, you could send it to me in email.

**Comment by markwal** [comment]
Oops.  Meant to @chrisheacock

**Comment by markwal** [comment]
OK.  I'm able to reproduce this and will fix in 1.1.2.  Need to test my fix on python 2.

**Comment by therealbstern** [comment]
Any updates or workarounds?

**Comment by Flying-Toast** [comment]
#12 Should fix this

**Comment by markwal** [comment]
Fixed in 1.1.2
